thunder/tests/test_nvfuser.py

```diff
@@ -961,3 +961,33 @@ def foo(t, ab):
     out_ref = foo(t, ab)
 
     assert out.equal(out_ref)
+
+
+# TODO: we should improve our consistency testing
+# to also include checks for the result of meta functions.
+@instantiate(
+    dtypes=(thunder.int64, thunder.int32),
+    executors=(nvFuserExecutor,),
+)
+def test_div_truediv_integer_tensors_consistency_nvfuser(executor, device, thunder_dtype):
+    dtype = ltorch.to_torch_dtype(thunder_dtype)
+
+    def div(a, b):
+        return thunder.prims.div(a, b)
+
+    def truediv(a, b):
+        return a // b
+
+    def make_integer_tensor():
+        half_len = 5
+        t = torch.tensor([*range(-half_len, 0), *range(1, half_len + 1)], device=device, dtype=dtype)
+        perm = torch.randperm(2 * half_len)
+        return t[perm]
+
+    x = make_integer_tensor()
+    y = make_integer_tensor()
+
+    for f in (thunder.jit(div), thunder.jit(truediv)):
+        rout = f(x.cpu(), y.cpu()).to(device)
+        jout = f(x, y)
+        assert rout.equal(jout)
```

Comment:
I don't think these results should be equal (I think we're exposing a bug in the nvFuser implementation of `prims.div`). `prims.div` should implement truncation division, which is not the same as Python's floor division. See, for example, the PyTorch executor's implementation of prims.div:

https://github.com/Lightning-AI/lightning-thunder/blob/44d2f3c9f02166945b2d737b42d86f28e4133b65/thunder/executors/torchex.py#L907

Reply:
fyi @kevinstephano @jjsjann123

Reply:
`div` calls to `prims.div`, so they should be equivalent.

Reply:
Yes, but doesn't `truediv` call floor division?

Reply:
Floor division has a decomp that is more that just `prims.div`

Reply:
The goal for now is to be able to handle integer tensors. I plan a follow-up to address all dtypes and consistency in general. Currently, our OpInfos skip NVFuser tests plus vjp.

Reply:
I understand and I'm not arguing for a high priority change, I'd just like to get us on the same page about this detail.

When `prims.div(a, b)` is called then it should perform a truncation division.

When `a // b` is written then it should perform a floor division.

And truncation division and floor division are different operations.

That's why I'm suggesting that this test is exposing an issue. Does that make sense? Maybe a quick VC would be simpler.

Reply:
They are equivalent for integer tensors, no?

Reply:
I am confused -- I am not comparing these two operations head to head, I just make sure that the cpu version is consistent with the nvfuser version for both `prims.div` and `* // *`.

Reply:
No, but it's a common misconception that they are. If the result is negative then they round differently. Floor division rounds door, truncation division rounds towards zero. 

In Python

```
-9 // 5
-2
```

In C++

```
#include <iostream>

int main() {
    int a = (-9 / 5);
    std::cout << a; 
}
```

prints -1. This is because -9 / 5 is -1.8, and floor division takes the floor of -1.8, which is -2, while truncation division truncates the fractional part of -1.8, giving -1.

Reply:
> fyi @kevinstephano @jjsjann123

Aha! @IvanYashchuk told me that nvFuser's div is performing truncation division for integers now. I mistakenly thought it wasn't. That's why this test passes

Reply:
while we are looking at torchex, I think the div prim impl has a typo (twice a instead of a and b) in the if condition:
https://github.com/Lightning-AI/lightning-thunder/blob/44d2f3c9f02166945b2d737b42d86f28e4133b65/thunder/executors/torchex.py#L908

Reply:
> while we are looking at torchex, I think the div prim impl has a typo (twice a instead of a and b) in the if condition:
> 
> https://github.com/Lightning-AI/lightning-thunder/blob/44d2f3c9f02166945b2d737b42d86f28e4133b65/thunder/executors/torchex.py#L908

I think you're right!